konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/api/pipeline/PipelineExecutor.java

```diff
@@ -76,4 +78,9 @@ default void close() {
      * Get the logger used by this PipelineExecutor
      */
     Logger getLogger();
+
+    /**
+     * Profiling
+     */
+    void profilerConfig(Profiler profiler);
```

Comment:
`void profilerConfig(ProfilerConfig profiler);`

Reply:
Still needs to be changed. User provides the config, we instantiate the Profiler instance internally.

Reply:
Where should we create Profiler object in snippet below? It depends on profilerConfig, so it must be created in pe.exec() method. Is it always safe? I'd move
 creation of profiler to PipelineExecutor constructor.

```
PipelineExecutor pe = p.executor();
pe.profilerConfig(new ProfilerConfig());
pe.exec(d);
```